include/libswiftnav/signal.h

```diff
@@ -61,6 +80,10 @@ static inline bool sid_is_equal(const gnss_signal_t a, const gnss_signal_t b)
   return sid_compare(a, b) == 0;
 }
 
+/** Constructs and returns a gnss_signal_t structure using given parameters.
+ */
+gnss_signal_t construct_sid(enum code code, u16 sat);
```

Comment:
Yes, when we start tracking for L2C from acquisition on L1CA we need to construct new sid and we know sat and code. So, for cases like this.